routing/pathfind.go

```diff
@@ -36,6 +36,11 @@ const (
 	// some effect with smaller time lock values. The value may need
 	// tweaking and/or be made configurable in the future.
 	RiskFactorBillionths = 15
+
+	// estimatedNodeCount is used to preallocate the path finding structures
+	// to avoid resizing and copies. It should be number on the same order as
+	// the number of active nodes in the network.
+	estimatedNodeCount = 0
```

Comment:
Set this to an actual number. Worst case it will be the number of reachable nodes on the network, so maybe `10k` or something should be good?

Reply:
Also  name it `heapSize`?

Reply:
I did a quick experiment with this again on testnet. Set the source node in the queryroutes call to something that doesn't exist, so that it will explore a relatively large portion of the graph. Result:
```
2019-10-23 12:03:04.616 [DBG] CRTR: Pathfinding perf metrics debug: distance_size=690, heap_key_map_size=372, heap_slice_size=372
2019-10-23 12:03:04.616 [DBG] CRTR: Pathfinding perf metrics: nodes=690, edges=4841, time=86.268991ms
```
It seems that the two slices and map don't really grow that large in comparison to the number of edges that are explored. Also, I didn't see much difference in execution time.

We've been discussing setting this upper bound for a while now. There isn't a natural default for it and querying from the db is slow. We could make it adaptive based on the previous run, but in the end I doubt whether it all matters much. There is also still that other slice in the heap that isn't pre-sized in the current version of this PR.

I wouldn't mind dropping this commit really, considering what we know now.

Reply:
Given that mainnet is much larger, do you get the same results there?

Reply:
FWIW, I tested all changes using a (old) snapshot of mainnet with ~6000 nodes and these changes had measurable impact. Though I tested it with the channel cache changeset applied so smaller changes were more noticeable.

Reply:
Ok, then also preallocated the slice in the heap and set the constant to 10k?